### PR TITLE
dep(react-native): Update to 0.75.4

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: ios/app/app.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1001,6 +1005,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				CC = "";
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply ios/app/app.xcodeproj/project.pbxproj:1008]: pod install.

File: babel.config.js [hunk 2]
```diff
@@ -5,5 +5,9 @@ module.exports = {
             plugins: [ 'react-native-paper/babel' ]
         }
     },
-    plugins: [ 'optional-require' ]
+    plugins: [ 'optional-require',
+        [ '@babel/plugin-transform-private-methods', {
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply babel.config.js:11]: Added comment.

File: ios/app/PrivacyInfo.xcprivacy [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply ios/app/PrivacyInfo.xcprivacy]: Moved the file to app and now we only have one which was updated.

File: babel.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (saghul) [reply babel.config.js:12]: But that's not the RN repo. Does the RN tester app use it?

File: ios/app/PrivacyInfo.xcprivacy [hunk 4]
```diff
@@ -17,6 +17,8 @@
 			<string>NSPrivacyAccessedAPICategoryUserDefaults</string>
 			<key>NSPrivacyAccessedAPITypeReasons</key>
 			<array>
+				<string>1C8F.1</string>
```
Comment (saghul) [inline ios/app/PrivacyInfo.xcprivacy:20]: Why are these needed now?

File: babel.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Calinteodor) [reply babel.config.js:12]: Showing Recent Messages
TransformError [SyntaxError]: /Users/cchitu/Work/jitsi-meet/node_modules/react-native/Libraries/Debugging/DebuggingOverlayRegistry.js: Class private methods are not enabled. Please add `@babel/plugin-transform-private-methods` to your configuration.

File: ios/app/PrivacyInfo.xcprivacy (hunk 4 above)
Comment (Calinteodor) [reply ios/app/PrivacyInfo.xcprivacy:20]: These are defaults, it seems. 
https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401

File: babel.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Calinteodor) [reply babel.config.js:12]: Build fails because of that error.

File: ios/app/PrivacyInfo.xcprivacy (hunk 4 above)
Comment (saghul) [reply ios/app/PrivacyInfo.xcprivacy:20]: 👍

File: ios/app/app.xcodeproj/project.pbxproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (saghul) [reply ios/app/app.xcodeproj/project.pbxproj:1008]: If you remove it, does it come back?

File: babel.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (saghul) [reply babel.config.js:12]: aha!

File: ios/app/app.xcodeproj/project.pbxproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (Calinteodor) [reply ios/app/app.xcodeproj/project.pbxproj:1008]: Yes

[the remark this answers is not in the document]
Comment (Calinteodor) [reply ios/app/app.xcodeproj/project.pbxproj:1008]: I reverted every change from both project files and didn't have any effect.

[the remark this answers is not in the document]
Comment (saghul) [reply ios/app/app.xcodeproj/project.pbxproj:1008]: Ack.

**Review by saghul — APPROVED** [review]
